### PR TITLE
acceptance tests

### PR DESCRIPTION
**Pull request by individual-it**
first iteration of acceptance tests for the oauth app
should tick a couple of boxes in the test-plan https://github.com/owncloud/QA/blob/master/Server/Test_Plan_OAuth2.md

### CODE REVIEW

File: tests/acceptance/features/webUIOauth2/accessToken.feature [hunk 1]
```diff
@@ -17,8 +17,19 @@ Feature: obtaining an access token
...
 			|username|password|displayname|email       |
 			|user1   |1234    |User One   |u1@oc.com.np|
-		When the user sends and oath2 authorization request using the webUI
+		When the user sends an oath2 authorization request using the webUI
 		And the user logs in with username "user1" and password "1234" using the webUI after a redirect from the oauth2AuthRequest page
 		And the user authorizes the oauth app using the webUI
 		And the client app requests an access token
+		Then the client app should be able to download the file "lorem.txt" of "user1" using the access token for authentification
+
+	Scenario: receive access token when user is alredy loged in
```
Comment (phil-davis) [inline tests/acceptance/features/webUIOauth2/accessToken.feature]: s/alredy/already

File: tests/acceptance/features/bootstrap/Oauth2Context.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	 * @var Oauth2AuthRequestPage
+	 */
+	private $oauth2AuthRequestPage;
+	
+	private $clientId = 'xdXOt13JKxym1B1QcEncf2XDkLAexMBFwiT9j6EfhhHFJhs2KM9jbjTmf8JBXE69';
+	private $clientSecret = 'UBntmLjC2yYCeHwsyj73Uwo9TAaecAetRwMw0xYcvNL9yRdLSUi0hUAHfvCHFeFh';
+	private $redirectUriPort;
+	private $redirectUriHost = "localhost";
+	/**
+	 * socket ressource representing the client app
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Oauth2Context.php]: s/ressource/resource

File: tests/acceptance/features/bootstrap/Oauth2Context.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	 * @param Oauth2AuthRequestPage $oauth2AuthRequestPage
+	 * 
+	 * @return void
+	 */
+	public function __construct(Oauth2AuthRequestPage $oauth2AuthRequestPage) {
+		$this->oauth2AuthRequestPage = $oauth2AuthRequestPage;
+	}
+
+	/**
+	 * @When the user sends an oath2 authorization request using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Oauth2Context.php]: s/oath2/oauth2

File: tests/acceptance/features/bootstrap/Oauth2Context.php [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+	 * @Given the user has send an oath2 authorization request
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Oauth2Context.php]: s/oath2/oauth2
s/send/sent

File: tests/acceptance/features/bootstrap/Oauth2Context.php [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+	 * 
+	 * @return void
+	 */
+	public function theUserSendsAndOathAuthorizationRequestUsingTheWebui() {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Oauth2Context.php]: s/And/An

File: tests/acceptance/features/bootstrap/Oauth2Context.php [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+		];
+		$fullUrl = $this->featureContext->getBaseUrl() .
+				   '/index.php/apps/oauth2/api/v1/token';
+		$request = $client->createRequest('POST', $fullUrl, $options);
+		$response = $client->send($request);
+		$this->accessTokenResponse = json_decode($response->getBody()->getContents());
+	}
+
+	/**
+	 * @Then the client app should be able to download the file :file of :user using the access token for authentification
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Oauth2Context.php]: s/authentification/authentication

File: tests/acceptance/features/webUIOauth2/accessToken.feature [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		Given these users have been created:
+			|username|password|displayname|email       |
+			|user1   |1234    |User One   |u1@oc.com.np|
+		When the user sends an oath2 authorization request using the webUI
+		And the user logs in with username "user1" and password "1234" using the webUI after a redirect from the oauth2AuthRequest page
+		And the user authorizes the oauth app using the webUI
+		And the client app requests an access token
+		Then the client app should be able to download the file "lorem.txt" of "user1" using the access token for authentification
+
+	Scenario: receive access token when user is alredy loged in
```
Comment (phil-davis) [inline tests/acceptance/features/webUIOauth2/accessToken.feature]: s/alredy/already

**Comment by individual-it** [comment]
this should be now ready for review. if the general direction is approved I can go on and write more tests

**Comment by individual-it** [comment]
drone tests do not pass yet because they need new QA tarballs build with https://github.com/owncloud/core/pull/31341 and https://github.com/owncloud/core/pull/31344

File: tests/acceptance/features/bootstrap/Oauth2Context.php [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+		$environment = $scope->getEnvironment();
+		// Get all the contexts you need in this context
+		$this->featureContext = $environment->getContext('FeatureContext');
+		$this->webUIGeneralContext = $environment->getContext('WebUIGeneralContext');
+		$this->webUILoginContext = $environment->getContext('WebUILoginContext');
+		$this->redirectUriPort = $this->findAvaliablePort();
+	}
+
+	/**
+	 * finds an availibale network port
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Oauth2Context.php]: s/availibale/available

File: tests/acceptance/features/bootstrap/Oauth2Context.php [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+	 * 
+	 * @return int port number
+	 */
+	private function findAvaliablePort() {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Oauth2Context.php]: typo in function name also.

**Pull request merged**